### PR TITLE
fix: don't output extra newline with single open php tag

### PR DESCRIPTION
**Pull request by evilebottnawi**

### CODE REVIEW

**Comment by czosel** [comment]
Is this specified by a styleguide?

**Comment by evilebottnawi** [comment]
@czosel it is ugly, right now we have  (two empty newline)
```php
$a = 1;


```

After (one empty newline), prettier always output one empty line in end of file:
```php
$a = 1;

```

**Comment by czosel** [comment]
Ah, that makes sense - thanks for the clarification! :+1:

File: src/printer.js [hunk 1]
```diff
@@ -368,10 +368,14 @@ function printStatement(path, options, print) {
         return concat([
           "<?php",
           hardline,
-          path.call(childrenPath => {
-            return printLines(childrenPath, options, print);
-          }, "children"),
-          hardline
```
Comment (czosel) [inline src/printer.js]: Why not just do it with a one-line change?
```
...,
path.call(childrenPath => {
  return printLines(childrenPath, options, print);
}, "children"),
node.children.length ? hardline : ''
```

Comment (evilebottnawi) [reply src/printer.js]: @czosel avoid extra `call`, improve perf

Comment (czosel) [reply src/printer.js]: Hmm, that sounds like a microoptimization to me - the effect is (a) probably very small and (b) only present when you're formatting an empty php file. I'd prefer to optimize for readability.

Comment (evilebottnawi) [reply src/printer.js]: @czosel let's do this

**Comment by evilebottnawi** [comment]
@czosel done

File: src/printer.js [hunk 2]
```diff
@@ -368,10 +368,12 @@ function printStatement(path, options, print) {
...
+          concat([
```
Comment (czosel) [inline src/printer.js]: I think you don't even need the extra call to `concat`

**Comment by evilebottnawi** [comment]
@czosel done :+1:

**Pull request merged**